lib/asset_cloud/base.rb

```diff
@@ -8,8 +8,7 @@ class IllegalPath < StandardError
 
   class Base
     cattr_accessor :logger
-
-    VALID_PATHS = /\A[a-z0-9][a-z0-9_\-\/]+([a-z0-9][\w\-\ \.\@]*\.\w{2,6})?\z/i
+    VALID_PATHS = /\A(([a-z0-9])|((\.?[\w\-\@])([\w\-\@]|[\ ][\w\-\@]|[\/][\w\-\@\.]|[\.][\w\-\@]+)*))\z/i
```

Comment:
space after slash was intentionaly rejected.
I hadn't thought of equals signs. Though I do suppose they are posix compliant.
space before dot was intentionally invalid
trailing period is a problem.
Colons are another character that is posix compliant but I didn't think of. Same with plusses.